src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -34,4 +37,10 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
   override def collectAsSeq[T <: WdlValue](filterFn: PartialFunction[WdlValue, T]): Seq[T] = {
     value flatMap { _.collectAsSeq(filterFn) }
   }
-}
\ No newline at end of file
+
+  override def getHash(implicit hasher: FileHasher) = {
+    val hashedArray = value map { _.getHash }
+    val accumulatedHash = hashedArray.foldLeft("") { (acc, v) => acc + v }
+    (getClass.getCanonicalName+accumulatedHash).md5Sum
+  }
```

Comment:
Hmm I don't think they will because `foldLeft` "folds" things from left to write so it should preserve the order.
Also there is a test for that in `WdlArraySpec` line 42.